### PR TITLE
Support `RouteInfo` in filters, add `createSpecific` to `ArgumentBinder`

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: core/src/main/java/io/micronaut/core/bind/ArgumentBinder.java [hunk 1]
```diff
@@ -47,6 +48,17 @@
...
 public interface ArgumentBinder<T, S> {
 
+    /**
+     * Create a specific binder.
+     *
+     * @param argument The bound argument
+     * @return The specific binder
+     * @since 4.8
+     */
+    default ArgumentBinder<T, S> createSpecific(Argument<T> argument) {
```
Comment (graemerocher) [inline core/src/main/java/io/micronaut/core/bind/ArgumentBinder.java]: nullability annotations

File: http-server/src/main/java/io/micronaut/http/server/binding/RouteInfoArgumentBinder.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * Argument binder for {@link io.micronaut.web.router.RouteInfo} objects.
+ *
+ * @since 4.8.0
+ * @author Denis Stepanov
+ */
+@Singleton
+@Internal
+public final class RouteInfoArgumentBinder implements TypedRequestArgumentBinder<RouteInfo<?>>, FilterArgumentBinderPredicate {
```
Comment (graemerocher) [inline http-server/src/main/java/io/micronaut/http/server/binding/RouteInfoArgumentBinder.java]: does it need to be public?

**Review by graemerocher — CHANGES_REQUESTED** [review]
Are there more argument binder that can benefit from createSpecific?

**Comment by dstepanov** [comment]
> Are there more argument binder that can benefit from createSpecific?

Considering it's in the core, other implementations can use it.

**Review by graemerocher — APPROVED** [review]

**Review by yawkat — APPROVED** [review]

**Pull request merged**